app/models/question.rb

```diff
@@ -244,6 +248,12 @@ def sorted_tags
     tags.order(:name)
   end
 
+  # Make audio prompt file name unique to curb collisions and duplications
+  def scramble_audio_prompt_filename
+    names = audio_prompt_file_name.split(".")
+    names.first + "_#{id.first(5)}.#{names.last}"
+  end
```

Comment:
This is pretty neat! 

I think you can simplify it more though. What about just 

```
"#{id}_audio_prompt#{File.extname(audio_prompt_file_name)}"
```

It's better to use `extname` since it's designed for this purpose. Also this method should go in a decorator in the ODK namespace since it's view logic.

Also as a rule, try to name your methods as nouns instead of verbs. So `scrambled_audio_prompt_filename` or `unique_audio_prompt_filename` might be better. (When I think of scrambling I think of mixing up the letters of something)

Reply:
👍

Reply:
I initially didn't put it in the decorator because I didn't want to initialize it with the form in the view